### PR TITLE
[4.x.x] Fix XML Catalogue Resolution

### PR DESCRIPTION
**Pull request by adamretter**
Previously we were using a custom OASIS Catalog Resolver, unfortunately this did not correctly resolve files in the catalogue that had relative path.

This replaces our custom code with @ndw's excellent xmlresolver.org library.

Many thanks to Norman Walsh for his help with this.

### CODE REVIEW

File: exist-core/src/main/java/org/exist/xquery/functions/validation/Jaxp.java [hunk 1]
```diff
@@ -345,14 +378,47 @@ private void setXmlReaderFeature(XMLReader xmlReader, String featureName, boolea
...
+            }
 
-        } catch (final SAXNotRecognizedException | SAXNotSupportedException ex) {
-            LOG.error(ex.getMessage());
+            try (final StringWriter stringWriter = new StringWriter()) {
+                final Properties outputProperties = new Properties();
+                outputProperties.setProperty(OutputKeys.METHOD, "XML");
+                outputProperties.setProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
+                outputProperties.setProperty(OutputKeys.INDENT, "no");
+                outputProperties.setProperty(OutputKeys.ENCODING, "UTF-8");
```
Comment (line-o) [inline exist-core/src/main/java/org/exist/xquery/functions/validation/Jaxp.java:395]: ehm... didn't we agree on using `UTF_8.name()`

**Comment by ndw** [comment]
Happy to help. Don't hesitate to reach out if you have questions.

I'll try to remember to let you know when I implement the "SaxProducer" interface you proposed for the resolver and Jing.

File: exist-core/src/main/java/org/exist/xquery/functions/validation/Jaxp.java (hunk 1 above)
Comment (adamretter) [reply exist-core/src/main/java/org/exist/xquery/functions/validation/Jaxp.java]: Yes we did, I missed that one. I have now corrected it :-)

Comment (line-o) [reply exist-core/src/main/java/org/exist/xquery/functions/validation/Jaxp.java]: Yeah, on the other hand I totally missed that this targets 4.x.x where the code still has the literal all over the place.

**Comment by line-o** [comment]
@adamretter do you plan to port this to 6.x.x ? I would like us to get this in.

**Comment by adamretter** [comment]
@line-o Yes, I will shortly send ports for 5.x.x and 6.x.x

**Review by dizzzz — APPROVED** [review]

**Pull request merged**